src/inmanta/execute/runtime.py

```diff
@@ -392,27 +403,37 @@ def __str__(self) -> str:
         return "BaseListVariable %s" % (self.value)
 
 
-class TempListVariable(BaseListVariable):
+class ListLiteral(BaseListVariable):
+    """
+    Transient variable to represent a list (of either constants or instances) literal (not a variable).
```

Comment:
The size of the list is fixed/known, which causes the promises tracking to be fully accurate (freezing is about guessing the size,...)

Reply:
Could you elaborate? I agree with what you're saying but I don't understand what you mean by the comment. As far as I can see it's just another way of saying what the docstring says.